### ISSUE TITLE
SSR without fetching data in server

### ISSUE THREAD

**Issue by zhaoyao91** [zeit/next.js#1033]
Could you please give some solutions or examples of doing ssr without fetching data in server? (with apollo).

For our cases, the auto code spliting and ssr for page structure are great and enough. We want our server only to deliver the minimum app.

Thanks.

**Comment by mikker** [comment]
Fetch the data inside of `componentDidMount()` instead of `getInitialProps()`.

The component doesn't _mount_ until it's rendered by the browser.

**Comment by sedubois** [comment]
You could also use <NoSSR> https://github.com/kadirahq/react-no-ssr

**Comment by sedubois** [comment]
Or just don't call await getDataFromTree

**Comment by arunoda** [comment]
Yep. I think these are the solutions.

**Comment by arunoda** [comment]
Check this example app: https://github.com/zeit/next.js/tree/master/examples/progressive-render

**Comment by zhaoyao91** [comment]
Thanks for all your replies. I've built my app according to [with-apollo example](https://github.com/zeit/next.js/blob/master/examples/with-apollo/lib/withData.js), and tried to implement this by 
1. only creating a singleton apollo client and redux store on the server.
2. removing all the `getInitialProps` code of `withData` HOC.
and the client seems to act as I expect, but I'm still curious about
1. how should I set the `ssrMode` option of apollo client?
2. Will this way ensure that the apollo client on the server will not try to fetch data?
3. Must I set ApolloProvider on the server? I think it's only useful on the client.

**Comment by sedubois** [comment]
1. I guess as long as you are client side, you can set ssr mode to false
2. If you don't call getDataFromTree, I guess the server code won't have the time to fetch any data. 
3. Can try to remove it but probably the Apollo code will complain it doesn't find the provider. I guess.

**Comment by zhaoyao91** [comment]
@sedubois I have struggled out a simple structure to fulfill the requirmemt. see [my app repo](https://github.com/zhaoyao91/our-poems)
I guess your 1 and 2 points are right. But when I removed the apollo provider on the server, it's the react on the client who complain that I've given different render code for ssr. So I have to mimic the components structure on the server, but I can confidently remove the data fetching code now.
